src/status_im2/contexts/syncing/events.cljs

```diff
@@ -52,27 +53,40 @@
 (rf/defn initiate-local-pairing-with-connection-string
   {:events       [:syncing/input-connection-string-for-bootstrapping]
    :interceptors [(re-frame/inject-cofx :random-guid-generator)]}
-  [{:keys [random-guid-generator db]} connection-string]
+  [{:keys [random-guid-generator db] :as cofx} connection-string]
   (let [installation-id (random-guid-generator)
         default-node-config (get-default-node-config installation-id)
         default-node-config-string (.stringify js/JSON (clj->js default-node-config))
         callback
         (fn [final-node-config]
-          (let [config-map (.stringify js/JSON
-                                       (clj->js
-                                        {:receiverConfig {:kdfIterations config/default-kdf-iterations
-                                                          :nodeConfig final-node-config
-                                                          :settingCurrentNetwork config/default-network
-                                                          :deviceType platform/os
-                                                          :deviceName
-                                                          (native-module/get-installation-name)}}))]
+          (let [config-map      (.stringify js/JSON
+                                            (clj->js
+                                             {:receiverConfig
+                                              {:kdfIterations config/default-kdf-iterations
+                                               :nodeConfig final-node-config
+                                               :settingCurrentNetwork config/default-network
+                                               :deviceType platform/os
+                                               :deviceName
+                                               (native-module/get-installation-name)}}))
+                includes-error? (fn [json-str]
+                                  (let [data (-> json-str
+                                                 js/JSON.parse
+                                                 (js->clj :keywordize-keys true))]
+                                    (let [error-value (get data :error)]
+                                      (and (not (nil? error-value)) (not (empty? error-value))))))]
             (rf/dispatch [:syncing/update-role constants/local-pairing-role-receiver])
             (native-module/input-connection-string-for-bootstrapping
              connection-string
              config-map
-             #(log/info "Initiated local pairing"
-                        {:response %
-                         :event    :syncing/input-connection-string-for-bootstrapping}))))]
+             (fn [res]
+               (signals/handle-local-pairing-signals
```

Comment:
This function needs to be abstracted so that it's not only specified for usage only by signals.